kolibri/plugins/learn/assets/src/modules/recommended/handlers.js

```diff
@@ -77,13 +78,39 @@ function _showLibrary(store, query, channels, baseurl) {
 }
 
 export function showLibrary(store, query, deviceId = null) {
-  if (deviceId) {
-    return setCurrentDevice(deviceId).then(device => {
-      const baseurl = device.base_url;
-      return fetchChannels({ baseurl }).then(channels => {
-        return _showLibrary(store, query, channels, baseurl);
-      });
+  /**
+   * ToDo: remove if block.
+   * Currently the channels & contentnode browser apis in studio
+   * are not able to load content using the the studio base url.
+   * Once studio is updated, this function will need to be refactored
+   * to use the else block code only.
+   *
+   * The if block is meant for UI viualization purposes only
+   * during development
+   */
+  if (deviceId === KolibriStudioId) {
+    RemoteChannelResource.getKolibriStudioStatus().then(({ data }) => {
+      if (data.status === 'online') {
+        RemoteChannelResource.fetchCollection().then(channels => {
+          //This is a hack to return kolibri channels.
+          store.commit('SET_ROOT_NODES', channels);
```

Comment:
maybe I don't understand how this should work, but the result are different of what I expected: 
![image](https://user-images.githubusercontent.com/1008178/224798355-fe4db9ee-3351-4905-817f-8d29008744ff.png)
The explored device has only one channel, as seen in the network request, however `rootNodes` is set to 10 channels that I don't know where they come from, but not from the explored device.

Reply:
@jredrejo I suspect that this is due to line 114, in the else case. The current implementation returns the previously loaded library channels as the newly selected library loads. This is an issue that should be resolved once loading states are streamlined across the remote content browsing feature.

Reply:
yep, it's showing both the channels existing in the local device plus the one on the remote device.

Reply:
Though the behavior is for it to be cleared once the loading is complete for remote devices. Not very sure the behavior with locally hosted devices